homeassistant/components/prometheus/__init__.py

```diff
@@ -807,6 +808,21 @@ def _handle_update(self, state: State) -> None:
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
+    def _handle_alarm_control_panel(self, mode: State) -> None:
+        current_mode = mode.state
+
+        if current_mode and VALID_STATES:
+            metric = self._metric(
+                "alarm_control_panel_state",
+                prometheus_client.Gauge,
+                "Alarm control panel state",
+                ["mode"],
+            )
+            for state in VALID_STATES:
+                metric.labels(**dict(self._labels(mode), mode=state)).set(
+                    float(state == current_mode)
+                )
```

Comment:
Sorry, I've realized that my previous comment about this was confusing. What we are reporting here seem to be referred to as "state" rather than "mode" by the `alarm_control_panel` component, so I would suggest to avoid using the term "mode". Here's a suggestion (you'd need to adjust tests accordingly):

```suggestion
    def _handle_alarm_control_panel(self, state: State) -> None:
        current_state = state.state

        if current_state and VALID_STATES:
            metric = self._metric(
                "alarm_control_panel_state",
                prometheus_client.Gauge,
                "State of the alarm control panel (0/1)",
                ["state"],
            )
            for s in VALID_STATES:
                metric.labels(**dict(self._labels(state), state=s)).set(
                    float(s == current_state)
                )
```